File: Backe/api/src/main/java/team/api/entity/Room.java
package team.api.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "rooms")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "room_id")
    private Integer roomId;

    @Column(nullable = false, length = 50)
    private String name;

    @Column(name = "total_seats")
    private Integer totalSeats;

    @Enumerated(EnumType.STRING)
    @Column(columnDefinition = "enum('active', 'maintenance') default 'active'")
    @Builder.Default
    private Status status = Status.active;

    public enum Status {
        active, maintenance
    }
}

File: Backe/DotnetBackend/Entities/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebPhimApi.Entities;

[Table("rooms")]
public class Room
{
    [Key]
    [Column("room_id")]
    public int RoomId { get; set; }

    [Required]
    [MaxLength(50)]
    public string Name { get; set; } = null!;

    [Column("total_seats")]
    public int? TotalSeats { get; set; }

    [Column(TypeName = "enum('active','maintenance')")]
    public RoomStatus Status { get; set; } = RoomStatus.active;

    // Navigation
    public ICollection<Seat> Seats { get; set; } = [];
    public ICollection<Showtime> Showtimes { get; set; } = [];
}

public enum RoomStatus { active, maintenance }
